### DISCUSSION TITLE
Run PackerCompile automatically when I edit and save my personal config.lua file

### DISCUSSION THREAD

**Discussion by c02y** [LunarVim/LunarVim#3965] [Q&A] [answered]
Sometimes after I edit and save my personal config.lua file, and open another lvim session, the config I make doesn't work, I have to run `PackerCompile` and reopen it, this issue occurs multiple times for me.

And I notice that if edit and save config.lua, run `PackerCompile`, it does nothing, I have to reopen lvim to run the command.

Doesn't anyone get annoyed by this stupid strategy?

Is it possible that after I edit and save my config.lua, lvim will automatically make the config work when I open a new lvim?
Such as 
- if I add a new plugin, new lvim session will automatically install it, 
- if I remove a plugin, new lvim session will automatically uninstall it, 
- if I change the config like the plugin's setup part, new lvim session will automatically make the change work?

**Comment by dominik-lukacs** [comment]
The annoying part to me isn't even having to run a command but that I have to exit out of lvim whenever I edit my config. Hope someone can provide a fix for that.

**Reply by c02y** [reply]
Indeed, https://github.com/LunarVim/LunarVim/commit/ccb80e41ee929505c97e682749c74f8db4609d61 it is going to use lazy.nvim, don't know if lazy.nvim needs this step.

**Reply by LostInTheLogs** [reply]
> The annoying part to me isn't even having to run a command but that I have to exit out of lvim whenever I edit my config. Hope someone can provide a fix for that.

neovim doesn't really support hot reload, we're doing what we can

> Indeed, https://github.com/LunarVim/LunarVim/commit/ccb80e41ee929505c97e682749c74f8db4609d61 it is going to use lazy.nvim, don't know if lazy.nvim needs this step.

there's no need for compiling in lazy.nvim

**Comment by LostInTheLogs** [comment, marked as answer]
> if I add a new plugin, new lvim session will automatically install it,

this is already a feature (no need to restart, just save the config)

> if I remove a plugin, new lvim session will automatically uninstall it,

this is a feature on master (no need to restart, just save the config)

> if I change the config like the plugin's setup part, new lvim session will automatically make the change work?

that's how lazy.nvim works
